pf/internal/check/checks.go

```diff
@@ -78,26 +75,79 @@ func checkIDProperties(sink diag.Sink, info tfbridge.ProviderInfo, isPFResource
 	return nil
 }
 
-func resourceHasRegularID(resource shim.Resource, resourceInfo *tfbridge.ResourceInfo) (bool, string) {
+type resourceError struct {
+	token string
+	err   error
+}
+
+func (err resourceError) Error() string {
+	msg := fmt.Sprintf("Resource %s has a problem", err.token)
+	if err.err != nil {
+		msg += ": " + err.err.Error()
+	}
+	return msg
+}
+
+type errSensativeID struct {
+	token string
+}
+
+func (err errSensativeID) Error() string {
+	msg := `"id" attribute is sensitive, but cannot be kept secret.`
```

Comment:
Would it be worth it to say why it cannot be kept secret?

Reply:
The best explanation for why is https://github.com/pulumi/pulumi/pull/8948, but it's not a very good explanation. I'll open an issue.

Reply:
https://github.com/pulumi/pulumi/issues/16818 is a slightly better explanation.